Mathlib/RingTheory/Congruence.lean

```diff
@@ -134,6 +134,7 @@ section Basic
 variable [Add R] [Mul R] (c : RingCon R)
 
 /-- Defining the quotient by a congruence relation of a type with addition and multiplication. -/
+@[reducible]
```

Comment:
Aside from this change, it's not clear to me that any of the other changes in this file are helpful; at the very least we should delete the comments if we do want the changes, but I'd be inclined to drop the them.

Reply:
Due to the indexing in the discrimination tree, `RingCon` will reduce to `Con` for the key, never to `AddCon`. Thus, all the instances on `AddCon` are filtered out immediately out on matching. `no_index` isn't really an option because that would make it match too often.